### DISCUSSION TITLE
Setting relative Offsets and Zeroing

### DISCUSSION THREAD

**Discussion by Festivejelly** [kachurovskiy/nanoels#157] [Q&A]
Given we can set a softlimit/endstop value ahead of the current location (also same applies for the zeroing)

Are we able to reverse the direction? So if I wanted to set the value behind my current location can I reverse the number I enter?

so if I enter 1.5mm how can I make that a minus? If its not possible maybe if we enter the number combined with 
![239652956-8eee528a-b628-4207-99b1-6f61b85c8be7](https://github.com/kachurovskiy/nanoels/assets/2408274/96af12af-3df1-4cf5-92f8-87c42093e1ee)

**Comment by kachurovskiy** [comment]
Nicholas, IIUC that would allow positioning yourself outside of soft limits which would result in confusing behavior like "I can't move in any direction because both left and right is technically outside of limits".

**Reply by Festivejelly** [reply]
Ah good point. I didnt think of that. Im just trying to think of a way of making to easier to enter the values for the soft limits for things like auto turning.

Ideally id rather just enter the desired diameter and length that I want. And keep the soft limits as the limits of travel rather than them being used for the cutting params.

I guess this can be achieved by working out what you need to set the soft limit for depth and lengths need to be but its all extra work :)

**Reply by Festivejelly** [reply]
@kachurovskiy I notice that sometimes when im trying to zero on the centreline it doesnt actually seem to apply that to the X co ordinate.

So for example if I cut a piece and measure it as 9.44mm if I enter 9.44 then press A then the X co ordinate changes relative to what it was. Rather than being set to 4.22. Curious why this would be?

**Reply by Festivejelly** [reply]
Okay this is maybe a bug.. its because I had not set the pitch first. As it was 0. After setting pitch, and applying with the play button. The above behaviour works correctly. Its probably mentioned in the docs tbh but it wasnt super clear.

**Reply by kachurovskiy** [reply]
Yes, there is a bug with `A` when `X` is not at `0`. Saw it the other day but couldn't reproduce. Thanks for bringing it up!

**Reply by kachurovskiy** [reply]
It's fixed in https://github.com/kachurovskiy/nanoels/commit/483a95b58423813276ff35a83e7909f94f1ab990
